### ISSUE TITLE
Building with a custom toolchain fails on OS X

### ISSUE THREAD

**Issue by onponomarev** [bazelbuild/bazel#3859]
### Description of the problem / feature request / question:

1) Clone any hazel-based C++ project.
2) Create a custom cpp toolchain either following the instruction https://github.com/bazelbuild/bazel/wiki/Building-with-a-custom-toolchain 
or just by copying the standard tools/cpp/BUILD and applying the following patch

```
diff --git a/tools/cpp/BUILD b/tools/cpp/BUILD
index a693b7610..35d216993 100644
--- a/tools/cpp/BUILD
+++ b/tools/cpp/BUILD
@@ -30,7 +30,7 @@ filegroup(
 # protobuf.
 alias(
     name = "toolchain",
-    actual = "//external:cc_toolchain",
+    actual = "default-toolchain",
 )

``` 
3)` build --crosstool_top=//tools/cpp:toolchain`
...
and
...
```
C++ compilation of rule '//src/base:base' failed (Exit 1).
src/main/tools/process-wrapper-legacy.cc:56: "execvp(tools/cpp/osx_cc_wrapper.sh, ...)": No such file or directory
```

exactly as in the issue #538

### Environment info
* Operating System:

MaxOS Sierra 10.12.6

* Bazel version (output of `bazel info release`):

release 0.6.0-homebrew

* If `bazel info release` returns "development version" or "(@non-git)", please tell us what source tree you compiled Bazel from; git commit hash is appreciated (`git rev-parse HEAD`):

###  Have you found anything relevant by searching the web?

issue #538

### Anything else, information or logs or outputs that would be helpful?
(If they are large, please upload as attachment or provide link).

**Comment by aehlig** [comment]
@lberki @mhlopko for further triaging.

**Comment by lberki** [comment]
(need to shed load, I'm a little overwhelmed)

**Comment by mhlopko** [comment]
Does the issue persist after `bazel clean --expunge`?

**Comment by onponomarev** [comment]
Yes.

**Comment by mhlopko** [comment]
I believe this issue is fixed by 16ccc84. Complain if not please.
